sdk/deepbook-v3/src/transactions/governance.ts

```diff
@@ -13,65 +13,77 @@ export class GovernanceContract {
 		this.#config = config;
 	}
 
-	stake =
-		(pool: Pool, balanceManager: BalanceManager, stakeAmount: number) => (tx: Transaction) => {
-			const tradeProof = tx.add(this.#config.balanceManager.generateProof(balanceManager));
-			const baseCoin = this.#config.getCoin(pool.baseCoin);
-			const quoteCoin = this.#config.getCoin(pool.quoteCoin);
+	stake = (
+		pool: Pool,
+		balanceManager: BalanceManager,
+		stakeAmount: number,
+		tx: Transaction = new Transaction(),
```

Comment:
I think this pattern makes sense in the client, but for the contracts, we should keep the pattern that I added here before of returning a function that accepts transaction